src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -104,6 +104,10 @@ public void addPerson(Person person) {
         updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
     }
 
+    @Override
+    public void addPersonForEdit(Person person) {
+        addressBook.addPerson(person);
+    }
```

Comment:
May I ask why this is needed?

Reply:
For the regular addPerson, it will reset the filter to display all the contacts.
For a user who has filtered by a certain criteria, then edited a person in the list, it might be confusing if the list is suddenly updated, causing the person they were updating to disappear from view.

Therefore addPersonForEdit keeps the current filter. Though if the user has filtered by name, then the displayed list will change.
*Rename addPersonForEdit to AddPersonKeepFilter for clarity.